.travis.yml

```diff
@@ -1,20 +1,32 @@
+os:
+  - linux
+
 language: go
-sudo: true
 
 go:
   - 1.11.x
 
+env:
+  global:
+    - GOTFLAGS="-race"
+  matrix:
+    - BUILD_DEPTYPE=gx
+    - BUILD_DEPTYPE=gomod
+
+
+# disable travis install
 install:
-  - make deps
+  - true
 
 script:
   - bash <(curl -s https://raw.githubusercontent.com/ipfs/ci-helpers/master/travis-ci/run-standard-tests.sh)
 
+
 cache:
   directories:
     - $GOPATH/src/gx
+    - $GOPATH/pkg/mod
+    - /home/travis/.cache/go-build
```

Comment:
Does it make sense to cache these? According to the [documentation](https://docs.travis-ci.com/user/caching/#things-not-to-cache)
> Large files that are quick to install but slow to download do not benefit from caching

Reply:
It reduces build time (on my machine) from 4 seconds to 0.23 seconds and the cache weights 4.7MB.

On Travis build time with warm go-build cache is 55s and 45s without. So you are right, it is not worth it.

Reply:
After running some more tests, averages come down to 42s with go-build cache and 46s without. (55s was big outlier).

Most improtatnly, build/test times are 4.8s vs 15s.